packages/kbn-logging/src/logger.ts

```diff
@@ -30,44 +29,44 @@ export interface Logger {
    * @param message - The log message
    * @param meta -
    */
-  trace(message: string, meta?: LogMeta): void;
+  trace<Meta extends LogMeta = LogMeta>(message: string, meta?: Meta): void;
```

Comment:
I really like this approach. Gives us type safety while also ensuring that consumers can add their own fields easily as intended by the ECS spec.

Reply:
> Only maybe if we allow any keys in LogMeta?

This is certainly an option too, the only downside is that we aren't doing anything to encourage folks to put meta in the right place, so we risk duplicating fields in ECS.

i.e. It wouldn't prevent someone from doing:
```ts
log.warn('foo', { fooServerAddress: '...' });
```

When what we really want is for them to do:
```ts
log.warn('foo', { server: { address: '...' } });
```

It definitely requires more changes if we don't have a fallback to `[key: string]: any`, however the advantage is that it requires folks to be explicit about any non-ECS field they put in their meta.

I could go either way on this TBH... I had tried using `[key: string]: any` at first, but then when I did the generics approach there were fewer type errors than I expected, which is why I kept that implementation. I think it's mainly a question of DX vs our desire for consistency with ECS. If we think this is the type of thing that will prevent folks from using log meta, I'd rather relax the requirement (some logs is better than no logs). But if we think it'd be easy enough to communicate to contributors, it would be nice to ensure a higher degree of ECS compliance.

WDYT?

Reply:
> however the advantage is that it requires folks to be explicit about any non-ECS field they put in their meta.

Fair point. I'm not sure it prevents users from 
```typescript
interface MyMeta extends LogMeta {
  fooServerAddress: string;
}
log.warn('foo', { fooServerAddress: '...' });
```
but probably it enforces them to see what properties are declared on `LogMeta` interface. So I'm fine with the suggested approach.